lib/rspec/core/profiler.rb

```diff
@@ -0,0 +1,38 @@
+module RSpec
+  module Core
+    # @private
+    class NoProfiler
+      def example_groups
+        {}
+      end
+    end
+
+    # @private
+    class Profiler
+      Formatters.register self, :example_group_started, :example_group_finished, :example_started
```

Comment:
> Well, it's registered as a formatter, and I thought that was what your definition of a formatter was...given that I've previously named a class ExceptionFormatter and you asked me to rename since it wasn't registered as a formatter.

It's abusing the `FormatterLoader`, but you can't actually load this as formatter, I can specify the notifications another way if you'd prefer. I asked you to rename it because it *wasn't* a formatter, not because it didn't use the lo

> Regardless, why can't this callback handling be moved into the existing ProfilerFormatter? Seems like it would have full access to the data tracked by the callbacks without having to complicate config and/or notifications with it.

Because it's used in the `JSONFormatter` too, and the idea is to make it accesible to any formatter that wishes to have profile output.

> I don't understand. Which 2 formatters are using this? It's brand new anyway... I really don't understand why this can't be merged into the other one.

The logic isn't brand new, it's extrapolated from `ProfileNotification` as per the original PR, the values are used in both the `ProfileFormatter` (for command line output) and `JSONFormatter`. Ostensibly it could be used by any formatter.